src/Microsoft.DotNet.XHarness.InstrumentationBase.Xunit/DefaultAndroidEntryPoint.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Microsoft.DotNet.XHarness.TestRunners.Common;
+using Microsoft.DotNet.XHarness.TestRunners.Xunit;
+
+namespace Microsoft.DotNet.XHarness.DefaultAndroidEntryPoint.Xunit
+{
+    /// <summary>
+    /// Class that implements basic functionality for the entry point of
+    /// the Android test application.
+    ///
+    /// The minimum required to run succesfully the test app based on
+    /// DefaultAndroidEntryPoint is to provide results-file-path as an argument.
+    ///
+    /// Client is able to provide test assemblies by overriding
+    /// GetTestAssemblies method.
+    ///
+    /// Other methods such as Device, Logger etc. have default implementation
+    /// but can be overrided if needed.
+    ///
+    /// </summary>
+    public class DefaultAndroidEntryPoint : AndroidApplicationEntryPoint
+    {
+        private const string ResultsFileArgumentName = "results-file-name";
+        private const string ResultsFileArgumentPath = "results-file-path";
+        private const string ExcludeCategoriesDirArgumentName = "exclude-categories-dir";
+        private const string ExcludeCategoriesFileArgumentName = "exclude-categories-file";
+        private const string ExcludeMethodArgumentName = "exclude-method";
+        private const string IncludeMethodArgumentName = "include-method";
+        private const string ExcludeClassArgumentName = "exclude-class";
+        private const string IncludeClassArgumentName = "include-class";
+        private readonly string _resultsPath;
+        private readonly string _excludeCategoriesDir;
+        private readonly string _excludeCategoriesFile;
+        private readonly Dictionary<string, string> _parsedArguments;
+
+        protected override string IgnoreFilesDirectory => _excludeCategoriesDir;
+        protected override string IgnoredTraitsFilePath => _excludeCategoriesFile;
+
+        public DefaultAndroidEntryPoint(Dictionary<string, string> bundle)
+        {
+            _parsedArguments = bundle;
+
+            // use default name for test results file
+            _parsedArguments.TryAdd(ResultsFileArgumentName, "TestResults.xml");
+
+            if (!Directory.Exists(_parsedArguments[ResultsFileArgumentPath]))
+            {
+                Directory.CreateDirectory(_parsedArguments[ResultsFileArgumentPath]);
+            }
+
+            _resultsPath = Path.Combine(_parsedArguments[ResultsFileArgumentPath], _parsedArguments[ResultsFileArgumentName]);
```

Comment:
There is no checks for this, so it will just crash if not provided. I'm thinking this should just be a string file path on the ctor.

Reply:
replaced